### PR TITLE
Add template catalog drawer

### PR DESCRIPTION
**Pull request by glekner**
## 📝 Description

- Added Initial Catalog Drawer + Test
- Added Empty state to Templates catalog
- Simplified Jest setup and Removed redundant `"@testing-library/dom"` for tests stability

## 🎥 Demo
Screen Shot 2022-02-10 at 16 19 15
Screen Shot 2022-02-10 at 16 19 08

### CODE REVIEW

**Comment by glekner** [comment]
please review @metalice @yaacov @avivtur @pcbailey

File: src/views/catalog/templatescatalog/TemplatesCatalog.tsx [hunk 1]
```diff
@@ -14,12 +17,23 @@ import { filterTemplates } from './utils/helpers';
 
 import './TemplatesCatalog.scss';
 
-const TemplatesCatalog: React.FC = () => {
+const TemplatesCatalog: React.FC<{
+  match?: {
```
Comment (metalice) [inline src/views/catalog/templatescatalog/TemplatesCatalog.tsx:21]: Consider TemplatesCatalogProps

File: src/views/catalog/templatescatalog/TemplatesCatalog.tsx [hunk 2]
```diff
@@ -62,22 +76,32 @@ const TemplatesCatalog: React.FC = () => {
...
-                    template={tmp}
-                    onClick={(temp) => console.log('clicked', temp?.metadata?.name)}
-                  />
-                ))}
-              </Gallery>
-            </StackItem>
+            {filteredTemplates?.length > 0 ? (
+              <StackItem className="co-catalog-page__grid vm-catalog-grid-container">
+                <Gallery hasGutter className="vm-catalog-grid">
+                  {filteredTemplates.map((tmp) => (
```
Comment (metalice) [inline src/views/catalog/templatescatalog/TemplatesCatalog.tsx:82]: consider temp => template or something more clear :)

File: src/views/catalog/templatescatalog/TemplatesCatalog.tsx [hunk 3]
```diff
@@ -14,12 +17,23 @@ import { filterTemplates } from './utils/helpers';
...
-const TemplatesCatalog: React.FC = () => {
+const TemplatesCatalog: React.FC<{
+  match?: {
+    params: {
+      ns: string;
+    };
+  };
+}> = ({
+  match: {
+    params: { ns },
```
Comment (metalice) [inline src/views/catalog/templatescatalog/TemplatesCatalog.tsx:28]: consider ns : namespace , better naming :)

File: src/views/catalog/templatescatalog/components/TemplatesCatalogDrawer/TemplateCatalogDrawer.scss [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  grid-column: 2;
+}
+
+.template-catalog-drawer-footer-tooltip {
+  display: flex;
+  align-items: center;
+}
+
+.template-catalog-drawer-form-name {
+  width: 250px;
```
Comment (metalice) [inline src/views/catalog/templatescatalog/components/TemplatesCatalogDrawer/TemplateCatalogDrawer.scss:28]: might be const? what about resizing?

File: src/views/catalog/templatescatalog/components/TemplatesCatalogDrawer/TemplatesCatalogDrawer.tsx [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+  getTemplateSupportLevel,
+} from '../../utils/helpers';
+import { getTemplateOSIcon } from '../../utils/os-icons';
+
+import { TemplatesCatalogDrawerFooter } from './TemplatesCatalogDrawerFooter';
+import { TemplatesCatalogDrawerPanel } from './TemplatesCatalogDrawerPanel';
+
+import './TemplateCatalogDrawer.scss';
+
+export const TemplatesCatalogDrawer: React.FC<{
```
Comment (metalice) [inline src/views/catalog/templatescatalog/components/TemplatesCatalogDrawer/TemplatesCatalogDrawer.tsx:20]: consider type TemplatesCatalogDrawerProps

File: src/views/catalog/templatescatalog/components/TemplatesCatalogDrawer/TemplatesCatalogDrawer.tsx [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+  namespace: string;
+  template: V1Template | undefined;
+  isOpen: boolean;
+  onClose: () => void;
+}> = React.memo(({ namespace, template, isOpen, onClose }) => {
```
Comment (metalice) [inline src/views/catalog/templatescatalog/components/TemplatesCatalogDrawer/TemplatesCatalogDrawer.tsx:25]: I know you passionate about React.memo , do we really need it? :)

File: src/views/catalog/templatescatalog/components/TemplatesCatalogDrawer/TemplatesCatalogDrawer.tsx [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+    <Modal
+      aria-label="template drawer"
+      className="ocs-modal co-catalog-page__overlay co-catalog-page__overlay--right template-catalog-drawer"
+      isOpen={isOpen}
+      onClose={onClose}
+      header={
+        template && (
+          <CatalogItemHeader
+            className="co-catalog-page__overlay-header"
+            title={getTemplateName(template)}
```
Comment (metalice) [inline src/views/catalog/templatescatalog/components/TemplatesCatalogDrawer/TemplatesCatalogDrawer.tsx:40]: adding it same as line 28?

File: src/views/catalog/templatescatalog/components/TemplatesCatalogDrawer/TemplatesCatalogDrawer.tsx [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+                <SplitItem>
+                  <strong>{t('Support level: ')}</strong> {support || 'N/A'}
+                </SplitItem>
+                <SplitItem> | </SplitItem>
+                <SplitItem>
+                  <strong>{t('Supports quick create VM')}</strong>
+                </SplitItem>
+              </Split>
+            }
+            iconImg={getTemplateOSIcon(template)}
```
Comment (metalice) [inline src/views/catalog/templatescatalog/components/TemplatesCatalogDrawer/TemplatesCatalogDrawer.tsx:56]: adding it same as line 28?

File: src/views/catalog/templatescatalog/components/TemplatesCatalogDrawer/TemplatesCatalogDrawer.tsx [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+            onCreate={(tmp) => console.log(tmp)}
+            onCancel={onClose}
+          />
+        )
+      }
+    >
+      {template && <TemplatesCatalogDrawerPanel template={template} />}
+    </Modal>
+  );
+});
```
Comment (metalice) [inline src/views/catalog/templatescatalog/components/TemplatesCatalogDrawer/TemplatesCatalogDrawer.tsx:74]: i can see in this file 3 times template && , consider something like const obj = template ? {footter: React.FC, ...} {footer: null, ...}

File: src/views/catalog/templatescatalog/components/TemplatesCatalogDrawer/TemplatesCatalogDrawerFooter.tsx [hunk 10]
```diff
@@ -0,0 +1,132 @@
...
+
+export const TemplatesCatalogDrawerFooter: React.FC<{
+  namespace: string;
+  template: V1Template | undefined;
+  onCreate: (template: V1Template) => void;
+  onCancel: () => void;
+}> = React.memo(({ namespace, template, onCreate, onCancel }) => {
+  const { t } = useTranslation('plugin__kubevirt-plugin');
+  const [vmName, setVmName] = React.useState(generateVMName(template));
+  const canQuickCreate = true;
```
Comment (metalice) [inline src/views/catalog/templatescatalog/components/TemplatesCatalogDrawer/TemplatesCatalogDrawerFooter.tsx:34]: it will rerender each prop change?

Comment (metalice) [reply src/views/catalog/templatescatalog/components/TemplatesCatalogDrawer/TemplatesCatalogDrawerFooter.tsx:34]: is it temp solution?

File: src/views/catalog/templatescatalog/components/TemplatesCatalogDrawer/TemplatesCatalogDrawerPanel.tsx [hunk 11]
```diff
@@ -0,0 +1,187 @@
...
+                </StackItem>
+              </Stack>
+            </div>
+          </div>
+        </div>
+      </div>
+    );
+  },
+);
+TemplatesCatalogDrawerPanel.displayName = 'TemplatesCatalogDrawerPanel';
```
Comment (metalice) [inline src/views/catalog/templatescatalog/components/TemplatesCatalogDrawer/TemplatesCatalogDrawerPanel.tsx:187]: please sperate each component to is own file :)

File: src/views/catalog/templatescatalog/components/TemplatesCatalogDrawer/TemplatesCatalogDrawerFooter.tsx (hunk 10 above)
Comment (glekner) [reply src/views/catalog/templatescatalog/components/TemplatesCatalogDrawer/TemplatesCatalogDrawerFooter.tsx:34]: temp

**Pull request merged**